Automattic/jetpack
# CLI: Fix issue where publicize command errored with bad data

Somehow, a test site ended up in a state where `$connections_to_return` looked like: `array( array( array( 'id' => 0, 'service' => 0 ) ) )`. This caused the CLI command to error when running `WP_CLI\Utils\format_items()` to output the data. 

#### Changes proposed in this Pull Request:

* To minimize similar issues happening in the future, this PR will validate `$connections_to_return` and will remove any connections that don't have the required keys.

#### Testing instructions:



- Connect Jetpack site
- Ensure there are no connection
- Run `wp jetpack publicize list` and ensure there is no output
- Connect a service
- Run `wp jetpack publicize list` and ensure that there is one connection in a table
- Run `wp shell` and then `Jetpack_Options::update_option( 'publicize_connections', array( array( array( 'id' => 0, 'service' => 0 ) ) ) );`
- Run `wp jetpack publicize list` and ensure that there is no output


#### Proposed changelog entry for your changes:
Adds some validation to minimize errors when running wp jetpack publicize.
